apps/meteor/client/components/GenericModal/GenericModal.tsx

```diff
@@ -21,6 +21,7 @@ type GenericModalProps = RequiredModalProps & {
 	tagline?: ReactNode;
 	onCancel?: () => Promise<void> | void;
 	onClose?: () => Promise<void> | void;
+	onDismiss?: () => Promise<void> | void;
```

Comment:
In which scenario is `onDismiss` necessary?

Reply:
The onClose was being used in the cleanup effect, this was causing problems with opening a modal from another modal because the onClose was being called and it was preventing the second from being displayed.

To avoid that scenario we introduced the onDismiss that is used in the cleanup effect if provided.